src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -348,21 +358,7 @@ public void RefreshRenameSessionWithOptionsChanged(SymbolRenameOptions newOption
             VerifyNotDismissed();
 
             _options = newOptions;
-
-            var cancellationToken = _cancellationTokenSource.Token;
-
-            UpdateReferenceLocationsTask(_threadingContext.JoinableTaskFactory.RunAsync(async () =>
-            {
-                // Join prior work before proceeding, since it performs a required state update.
-                // https://github.com/dotnet/roslyn/pull/34254#discussion_r267024593
-                //
-                // The cancellation token is passed to the prior work when it starts, not when it's joined. This is
-                // the equivalent of TaskContinuationOptions.LazyCancellation.
-                await _allRenameLocationsTask.JoinAsync(CancellationToken.None).ConfigureAwait(false);
-                await TaskScheduler.Default;
-
-                return await _renameInfo.FindRenameLocationsAsync(_options, cancellationToken).ConfigureAwait(false);
-            }));
```

Comment:
instead of having two callers with slightly different logic (but both of which need to do `await _renameInfo.FindRenameLocationsAsync(...)`, we now call into one consistent helper which determines teh right course of action based on if this is the first time we're kicking work off, or if we're kicking work off based on an options change.